B-教学案例与实践/B13-AI对联生成案例/src/training/how_to_train.md

```diff
@@ -0,0 +1,96 @@
+# 环境配置
+
+Ubuntu 16.04
+
+Anaconda3 安装 python3.x 环境，或本机 Python3.x 环境
+
+安装 training/train_requirements.txt 中所需要的 python packages
+
+
+# 数据收集
+
+1. 从 GitHub 网站下载 couplet v1.0 release 版本的数据，并解压。
+
+    **GitHub网站**：https://github.com/wb14123/couplet-dataset/releases
+
+    **文件名**：couplet.tar.gz
+
+2. 下载 Tensor2Tensor 库 v1.2.9 版本
+    ```
+    git clone https://github.com/tensorflow/tensor2tensor.git
+    git checkout v1.2.9
+    ```
+
+
+# 数据处理
+
+Couplet 数据解压后，有两个文件夹，分别存放 train 和 test 的上下联数据。 In 为上联， out 为下联。在训练过程中，我们只需对 train 的数据进行预处理。
+
+1. 创建 data 子目录。
+2. 将 couplet 中 train 和 test 目录下的数据移动至 data 目录下，并重命名为：
+    ```
+    train.in.txt，train.out.txt，test.in.txt，test.out.txt。
+    ```
+3. 在 data 目录中，合并上下联文件并保存结果：
+    ```
+    cat train.in.txt train.out.txt > train.merge.txt
+    ```
+4. 在data 目录中，统计字表数目并保存字表：
+
+    需要安装 subword-nmt 包：
+    ` ``
+    python -m pip install subword-nmt
```

Comment:
>subword-nmt [](start = 26, length = 11)

How about add this package to `train_requirements.txt`

Reply:
Sure. Added to the requirements file